x-pack/plugins/triggers_actions_ui/public/application/lib/alert_type_compare.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AlertTypeModel } from '../../types';
+import { IsEnabledResult, IsDisabledResult } from './check_alert_type_enabled';
+
+export function alertTypeSolutionCompare(
+  a: [
+    string,
+    Array<{
+      id: string;
+      name: string;
+      checkEnabledResult: IsEnabledResult | IsDisabledResult;
+      alertTypeItem: AlertTypeModel;
+    }>
+  ],
+  b: [
+    string,
+    Array<{
+      id: string;
+      name: string;
+      checkEnabledResult: IsEnabledResult | IsDisabledResult;
+      alertTypeItem: AlertTypeModel;
+    }>
+  ],
+  solutions: Map<string, string> | undefined
+) {
+  // .sort(([a], [b]) =>
+  // solutions ? solutions.get(a)!.localeCompare(solutions.get(b)!) : a.localeCompare(b)
+  // )
+  const solutionA = a[0];
+  const solutionB = b[0];
+  const alertTypeItemsA = a[1].find((alertTypeItem) => alertTypeItem.checkEnabledResult.isEnabled);
+  const alertTypeItemsB = b[1].find((alertTypeItem) => alertTypeItem.checkEnabledResult.isEnabled);
+
+  if (!!alertTypeItemsA && !alertTypeItemsB) {
+    return -1;
+  }
+  if (!alertTypeItemsA && !!alertTypeItemsB) {
+    return 1;
+  }
```

Comment:
The mix of `!!` and `!` makes this a little hard to follow... can we make this a little clearer?
A couple of suggestions that might help:
1. Can we rename `A` and `B` to something more descriptive? Perhaps `AlertTypesListA`? It's common to use `Left` and `Right` rather than `A` and `B`. So perhaps `LeftAlertTypesList` and `RightAlertTypesList`?
2. change `alertTypeItemsA` to a more descriptive name perhaps? Maybe `enabledAlertTypeInList`?
3. perhaps, to avoid the mix of `!!` with `!`, we do something like:

```
 const hasEnabledAlertTypeInListA = a[1].find((alertTypeItem) => alertTypeItem.checkEnabledResult.isEnabled) != undefined;
```

Something like that?